src/improvements/AddressRegistry.sol

```diff
@@ -289,6 +291,11 @@ contract AddressRegistry is StdChains {
         return chains;
     }
 
+    /// @notice reads a ChainInfo array from a toml file
+    function readChainsFromToml(string memory toml, string memory key) public view returns (ChainInfo[] memory) {
+        return abi.decode(vm.parseToml(vm.readFile(toml), key), (ChainInfo[]));
+    }
```

Comment:
We can even hardcode the `key` here since it will always be the same
```suggestion
    function readChainsFromToml(string memory toml) public view returns (ChainInfo[] memory) {
        return abi.decode(vm.parseToml(vm.readFile(toml), ".l2Chains"), (ChainInfo[]));
    }
```